corehq/util/view_utils.py

```diff
@@ -97,10 +97,10 @@ def _get_json_exception_response(code, request, exception, log_message=None):
 
 
 def _json_exception_response_data(code, exception):
-    if isinstance(exception.message, six.text_type):
-        message = six.text_type(exception)
+    if isinstance(exception.args[0], six.binary_type):
+        message = exception.args[0].decode('utf-8')
     else:
-        message = str(exception).decode('utf-8')
+        message = six.text_type(exception)
```

Comment:
would `six.text_type(exception)` not work for all exceptions here?

Reply:
It looks like that option was already considered:

https://github.com/dimagi/commcare-hq/pull/11198#issuecomment-208339023

And a test was added:

https://github.com/dimagi/commcare-hq/pull/11198/commits/d2222649ad8530a18c3b6e52ffe16f54100b58da

that would break with ```six.text_type(exception)```